cmd/thanos/store.go

```diff
@@ -279,11 +279,15 @@ func runStore(
 	}
 
 	// Ensure we close up everything properly.
-	defer func() {
-		if err != nil {
-			runutil.CloseWithLogOnErr(logger, bkt, "bucket client")
-		}
-	}()
+	{
+		done := make(chan struct{})
+		g.Add(func() error {
+			<-done
+			return bkt.Close()
```

Comment:
Should we retain `CloseWithLogOnErr` here so that we can log the potential close error?